### PR TITLE
Feature test

### PR DESCRIPTION
**Pull request by XiaoPa**
Updated:
- README.md with data input format
- Added 'postal_town' as one of input filed
- changed start/end symbol from `<s>` to`#` it helps to recognise more accurately
- Modified Json encoding when post data
- added four/five grams
- added data clean methods when tagging

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,3 +6,12 @@ In the project's folder run:
 ```
 docker-compose up -d
 ```
+
+Input data format
+```curl --data "data=[{\"_id\": \"stringIdOfRow1\", \"name\": \"Post Office Trafalgar Square\", \"postal_town\": \"London\"}, {\"_id\": \"stringIdOfRow2\", \"name\": \"Post Office\", \"postal_town\": \"London\"}, {\"_id\": \"stringIdOfRow3\", \"name\": \"Post Office\", \"postal_town\": \"London\"}, {\"_id\": \"stringIdOfRow4\", \"name\": \"Post Office\", \"postal_town\": \"London\"}, {\"_id\": \"stringIdOfRow5\", \"name\": \"Post Office\", \"postal_town\": \"London\"}]" http://127.0.0.1:5000/api/v1.0/datasort
+```
+or without ```"data=```
```
Comment (maZahaca) [inline README.md]: unnecessary "

File: README.md [hunk 2]
```diff
@@ -6,3 +6,12 @@ In the project's folder run:
...
+```
+curl --data "[{\"_id\": \"stringIdOfRow1\", \"name\": \"Post Office Trafalgar Square\", \"postal_town\": \"London\"}, {\"_id\": \"stringIdOfRow2\", \"name\": \"Post Office\", \"postal_town\": \"London\"}, {\"_id\": \"stringIdOfRow3\", \"name\": \"Post Office\", \"postal_town\": \"London\"}, {\"_id\": \"stringIdOfRow4\", \"name\": \"Post Office\", \"postal_town\": \"London\"}, {\"_id\": \"stringIdOfRow5\", \"name\": \"Post Office\", \"postal_town\": \"London\"}]" http://127.0.0.1:5000/api/v1.0/datasort
+
+```
```
Comment (maZahaca) [inline README.md]: What's about example of output?

File: src/data_sort.py [hunk 3]
```diff
@@ -52,20 +58,37 @@ def replace_no_char(strings):
...
 	def add_break(strings):
-		strings = '<s>'+ strings + '</s>'
+		strings = '#' + strings + '#'
 		return strings	    
 
+
+	#Return clean addresses for cleaning names
+	def get_clean_address(self):
+		dataframe = self.read_data()
+		if 'postal_town' in dataframe.columns:
```
Comment (maZahaca) [inline src/data_sort.py]: could you please change it to "town"

Comment (maZahaca) [reply src/data_sort.py]: Are you ready to the case if it's empty?

File: src/data_sort.py [hunk 4]
```diff
@@ -102,8 +126,8 @@ def delete_nonsense(count_dict):
 	@staticmethod
 	def tranformations(strings):
 		strings = ' '.join(strings)
-		strings = strings.replace('<s>','')
-		strings = strings.replace('</s>','')
+		strings = strings.replace('#','')
+		# strings = strings.replace('#','')
```
Comment (maZahaca) [inline src/data_sort.py]: Don't leave commented code, it's a bad practice

File: src/data_sort.py [hunk 5]
```diff
@@ -171,8 +212,8 @@ def tagging(self,target_df):
 	#create a column called tags
 	def get_tagged(self):
 		data = self.read_data()
-		data['tags'] =  data.name.apply(self.encode_data).apply(lambda x: x.lower()).apply(self.tagging)
-		# data.applymap(self.encode_data).to_csv('results.csv') this line is for local testing 
+		data['tags'] =  data.name.apply(self.encode_data).apply(lambda x: x.lower()).apply(self.replace_no_char).apply(self.replace_no_char_on_tag).apply(self.tagging)
+		#data.applymap(self.encode_data).to_csv('../../../results.csv') #this line is for local testing
```
Comment (maZahaca) [inline src/data_sort.py]: the same commented code block

**Review by maZahaca — CHANGES_REQUESTED** [review]

**Comment by maZahaca** [comment]
Next time please:
- create a new branch for each feature
- branch name should starts from "feature-" and contains the main meaning of what you're doing in it.

Roger that.

File: README.md (hunk 1 above)
Comment (XiaoPa) [reply README.md]: deleted

File: README.md (hunk 2 above)
Comment (XiaoPa) [reply README.md]: Updated.
Expected output format, if success:

```
{
    "results": [
      {
        "count": 5,
        "groupName_Id": 0,
        "rows": [
                "stringIdOfRow1",
                "stringIdOfRow2",
                "stringIdOfRow3",
                "stringIdOfRow4",
                "stringIdOfRow5"
                ],
        "tags": "post office"
      }
    ],
    "status": "success"
}
```

if error:

```
{
      "error": "string indices must be integers, not str",
      "status": "error"
}
```

File: src/data_sort.py (hunk 3 above)
Comment (XiaoPa) [reply src/data_sort.py]: in the case of empty, nothing will be executed.

File: src/data_sort.py (hunk 4 above)
Comment (XiaoPa) [reply src/data_sort.py]: updated

File: src/data_sort.py (hunk 5 above)
Comment (XiaoPa) [reply src/data_sort.py]: updated

**Pull request merged**